src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/TimeSpanExtensions.cs

```diff
@@ -72,5 +80,98 @@ public static TimeSpan CalculateBackoffDuration(this TimeSpan minDuration, TimeS
+        /// <returns>The backoff duration before retrying a request to the configuration store or replica again.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// An exception is thrown when <paramref name="attempts"/> is less than 1.
+        /// </exception>
+        public static TimeSpan CalculateExponentialStartupBackoffDuration(this TimeSpan minDuration, TimeSpan maxDuration, int attempts, TimeSpan startupTimeElapsed)
+        {
+            if (attempts < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(attempts), attempts, "The number of attempts should not be less than 1.");
+            }
+
+            if (attempts == 1)
+            {
+                return minDuration;
+            }
+
+            //
+            // IMPORTANT: This can overflow
+            double calculatedMilliseconds = Math.Max(1, minDuration.TotalMilliseconds) * ((long)1 << Math.Min(attempts, MaxAttempts));
+
+            if (calculatedMilliseconds > maxDuration.TotalMilliseconds ||
+                    calculatedMilliseconds <= 0 /*overflow*/)
+            {
+                calculatedMilliseconds = maxDuration.TotalMilliseconds;
+            }
+
+            TimeSpan maxBackoffDuration = CalculateMaxStartupBackoffDuration(startupTimeElapsed);
+
+            TimeSpan calculatedDuration = TimeSpan.FromMilliseconds(calculatedMilliseconds);
+
+            if (calculatedDuration > maxBackoffDuration)
+            {
+                calculatedDuration = maxBackoffDuration;
+            }
+
+            return calculatedDuration.Jitter(StartupJitterRatio);
+        }
+
+        /// <summary>
+        /// This method calculates the fixed backoff duration for the configuration store after a failure
+        /// during startup 
+        /// </summary>
+        /// <param name="startupTimeElapsed">The time elapsed since the current startup began.</param>
+        /// <returns>The backoff duration before retrying a request to the configuration store or replica again.</returns>
+        public static TimeSpan CalculateFixedStartupBackoffDuration(this TimeSpan startupTimeElapsed)
+        {
+            return CalculateMaxStartupBackoffDuration(startupTimeElapsed).Jitter(StartupJitterRatio);
+        }
+
+        private static TimeSpan Jitter(this TimeSpan timeSpan, double ratio)
+        {
+            if (ratio < 0 || ratio > 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(ratio));
+            }
+
+            if (ratio == 0)
+            {
+                return timeSpan;
+            }
+
+            var rand = new Random();
+
+            double interval = Math.Abs(timeSpan.TotalMilliseconds * ratio);
+
+            double lowest = timeSpan.TotalMilliseconds - interval * 0.5;
+
+            double offset = interval * rand.NextDouble();
+
+            return TimeSpan.FromMilliseconds(lowest + offset);
```

Comment:
For a ratio of 0.5, do we expect a jitted time of 10s falls in [5, 15]? If so, above implementation is incorrect as it generates the range of [7.5, 12.5]. I hope something like below is easier to understand:

```cs
double jitter = ratio * (rand.NextDouble() * 2 - 1);
double interval = timeSpan.TotalMilliseconds * (1 + jitter);
return TimeSpan.FromMilliseconds(interval);
```

Reply:
jittering a 10s duration by a ratio of .5 introduces a 5 second (10 x .5) window. So the expectation is [7.5, 12.5]

Reply:
The formula can be updated easily to achieve the desired results, but that's not intuitive to me. The 50% jitter doesn't give me 50% more (or less) than what I had. So if I want 100% more (or less) than what I had, I actually need a 200% jitter. Isn't that weird?

Reply:
Discussed separately with @zhenlan. Seems more people interpret the ratio here as the delta from the extreme end of the possible duration window to the original duration. In that case, we would benefit to change it.

So jitter(10, .5) yields [5, 15] rather than [7.5, 12.5]